src/test/java/com/yelp/nrtsearch/server/ExistsQueryTest.java

```diff
@@ -0,0 +1,164 @@
+            .putFields("name", MultiValuedField.newBuilder().addValue("").build())
+            .putFields("pickup_partners", MultiValuedField.newBuilder().addValue("[]").build())
+            .putFields("delivery_areas", MultiValuedField.newBuilder().addValue("[]").build())
+            .build();
+
+    // Should match all exists queries
+    Map<String, Object> pickup2 = Map.of("hours", List.of(2));
+    Map<String, Object> deliveryAreas2 = Map.of("partner", Map.of("partner_id", 123));
+
+    AddDocumentRequest fullBusiness =
+        AddDocumentRequest.newBuilder()
+            .setIndexName(name)
+            .putFields("id", MultiValuedField.newBuilder().addValue("2").build())
+            .putFields("name", MultiValuedField.newBuilder().addValue("restaurantA").build())
+            .putFields(
+                "pickup_partners",
+                MultiValuedField.newBuilder().addValue(gson.toJson(pickup2)).build())
+            .putFields(
+                "delivery_areas",
+                MultiValuedField.newBuilder().addValue(gson.toJson(deliveryAreas2)).build())
+            .build();
+
+    addDocuments(List.of(emptyBusiness, minimalBusiness, fullBusiness).stream());
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void testAddNullMultiValuedField() {
+    // Verify that clientlib doesn't allow null
+    AddDocumentRequest.newBuilder()
+        .setIndexName("test_index")
+        .putFields("id", MultiValuedField.newBuilder().addValue("0").build())
+        .putFields("name", MultiValuedField.newBuilder().addValue(null).build())
+        .build();
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void testAddNullField() {
+    // Verify that clientlib doesn't allow null
+    AddDocumentRequest.newBuilder()
+        .setIndexName("test_index")
+        .putFields("id", MultiValuedField.newBuilder().addValue("0").build())
+        .putFields("name", null)
+        .build();
+  }
+
+  @Test
+  public void testTopLevelField() {
+    // Empty strings become valid lucene fields
+    ExistsQuery existsQuery = ExistsQuery.newBuilder().setField("name").build();
+    queryAndVerifyIds(existsQuery, "1", "2");
+  }
+
+  // TODO:(samir|2020-01-21): Test exists query on nested object itself (Ex. pickup_partners)
+  @Test
+  public void testNestedField() {
+    Query nestedQuery =
+        Query.newBuilder()
+            .setNestedQuery(
+                NestedQuery.newBuilder()
+                    .setPath("pickup_partners")
+                    .setQuery(
+                        Query.newBuilder()
+                            .setExistsQuery(
+                                ExistsQuery.newBuilder().setField("pickup_partners.hours").build()))
+                    .build())
+            .build();
+
+    queryAndVerifyIds(nestedQuery, "2");
+  }
+
+  // TODO:(samir|2020-01-21): Test exists query on object itself
+  //  (Ex. delivery_areas or delivery_areas.partner)
```

Comment:
currently exists query may not work for object field. If object field is not stored or store dco value, document.getFields will not have it. We could handle they separately. later.

Reply:
Yeah I'm leaving the TODO here to test exists on the object itself. I'm not certain if ES even supports it, and I don't believe this will be a blocker for ads_search.